tests/lapi/bitop_arshift_test.lua

```diff
@@ -0,0 +1,28 @@
+--[[
+SPDX-License-Identifier: ISC
+Copyright (c) 2023-2025, Sergey Bronnikov.
+
+Synopsis: bit32.arshift (x, disp)
+]]
+
+local luzer = require("luzer")
+local test_lib = require("lib")
+
+if test_lib.version() ~= "LuaJIT" then
+    os.exit(0)
+end
+
+local bitop = require("bit")
+local arshift = bitop.arshift
+
+local function TestOneInput(buf)
+    local fdp = luzer.FuzzedDataProvider(buf)
+    local x = test_lib.random_number(fdp)
+    local y = test_lib.random_number(fdp)
+    arshift(x, y)
```

Comment:
Let's add the check that the sign of the result is the same as for `x` operand.

Reply:
Updated:

```diff
--- a/tests/lapi/bitop_arshift_test.lua
+++ b/tests/lapi/bitop_arshift_test.lua
@@ -14,6 +14,11 @@ if test_lib.lua_version() ~= "LuaJIT" then
 end
 
 local arshift = bit.arshift
+local bxor = bit.bxor
+
+local function is_opposite_sign(a, b)
+    return bxor(a, b) < 0
+end
 
 local function TestOneInput(buf)
     local fdp = luzer.FuzzedDataProvider(buf)
@@ -23,6 +28,7 @@ local function TestOneInput(buf)
     local n = fdp:consume_integer(MIN_INT, MAX_INT)
     local res = arshift(x, n)
     assert(type(res) == "number")
+    assert(is_opposite_sign(x, res) == false)
 end
 
 local args = {
```